apps/discord-bot/src/listeners/events/auto-thread.test.ts

```diff
@@ -0,0 +1,260 @@
+import { Client, Events, TextChannel } from "discord.js";
+import {
+  mockTextChannel,
+  mockMessage,
+  mockGuildMember,
+  emitEvent,
+} from "@answeroverflow/discordjs-mock";
+import { setupAnswerOverflowBot } from "~discord-bot/test/sapphire-mock";
+import { randomSnowflake } from "@answeroverflow/discordjs-utils";
+import { mockPublicThread } from "@answeroverflow/discordjs-mock";
+import { createChannel, createServer } from "@answeroverflow/db";
+import { toAOServer, toAOChannel } from "~discord-bot/utils/conversions";
+
+let client: Client;
+let textChannel: TextChannel;
+
+beforeEach(async () => {
+  client = await setupAnswerOverflowBot();
+  textChannel = mockTextChannel(client);
+});
+
+const succeedCreatingAThread = async () => {
+  // Channel with flag enabled
+  await createServer(toAOServer(textChannel.guild));
+  await createChannel({
+    ...toAOChannel(textChannel),
+    flags: { autoThreadEnabled: true },
+  });
+  const author = mockGuildMember({ client });
+  const message = mockMessage({
+    client,
+    channel: textChannel,
+    author: author.user,
+    override: {
+      content: "Message",
+    },
+  });
+
+  await emitEvent(client, Events.MessageCreate, message);
+
+  expect(message.startThread).toHaveBeenCalled();
+};
+
+describe("Auto thread", () => {
+  it("should not create a thread on a channel thread channel", async () => {
+    await succeedCreatingAThread();
+
+    const channel = mockPublicThread({ client });
+    const author = mockGuildMember({ client });
+    const message = mockMessage({
+      client,
+      channel: channel,
+      override: {
+        content: "Hey",
+      },
+      author: author.user,
+    });
+
+    await emitEvent(client, Events.MessageCreate, message);
+
+    expect(message.startThread).not.toHaveBeenCalled();
+  });
+  it("should not create a thread if the author is a bot", async () => {
+    await succeedCreatingAThread();
+    const author = mockGuildMember({
+      client,
+      data: {
+        user: {
+          bot: true,
+          id: randomSnowflake().toString(),
+        },
+      },
+    });
+
+    const message = mockMessage({
+      client,
+      channel: textChannel,
+      override: {
+        content: "Hey",
+      },
+      author: author.user,
+    });
+
+    await emitEvent(client, Events.MessageCreate, message);
+
+    // assert
+    expect(message.startThread).not.toHaveBeenCalled();
```

Comment:
Now that we're actually mocking the thread creation let's assert that the thread doesn't exist anymore instead of checking it the function has been called

Reply:
Yeah, it may fail since I'm not sure if we populate the cache or not, if it does fail we'll have to update DiscordJS-mock